dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/task/client/PhysicalTaskExecutorClientDelegator.java

```diff
@@ -21,26 +21,108 @@
+                        String.format("Cannot find the host to dispatch Task[id=%s, name=%s]",
+                                taskExecutionContext.getTaskInstanceId(), taskName)));
+
+        taskExecutionContext.setHost(physicalTaskExecutorAddress);
+        taskExecutionRunnable.getTaskInstance().setHost(physicalTaskExecutorAddress);
+
+        try {
+            final TaskExecutorDispatchResponse taskExecutorDispatchResponse = Clients
+                    .withService(IPhysicalTaskExecutorOperator.class)
+                    .withHost(physicalTaskExecutorAddress)
+                    .dispatchTask(TaskExecutorDispatchRequest.of(taskExecutionRunnable.getTaskExecutionContext()));
+            if (!taskExecutorDispatchResponse.isDispatchSuccess()) {
+                throw new TaskDispatchException(
+                        "Dispatch task: " + taskName + " to " + physicalTaskExecutorAddress + " failed: "
+                                + taskExecutorDispatchResponse);
+            }
+        } catch (TaskDispatchException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new TaskDispatchException(
+                    "Dispatch task: " + taskName + " to " + physicalTaskExecutorAddress + " failed", e);
+        }
+    }
+
+    @Override
+    public boolean reassignMasterHost(final ITaskExecutionRunnable taskExecutionRunnable) {
+        final String taskName = taskExecutionRunnable.getName();
+        checkArgument(taskExecutionRunnable.isTaskInstanceInitialized(),
+                "Task " + taskName + "is not initialized cannot take-over");
+
+        final TaskInstance taskInstance = taskExecutionRunnable.getTaskInstance();
+        final String taskExecutorHost = taskInstance.getHost();
+        if (StringUtils.isEmpty(taskExecutorHost)) {
+            log.debug(
+                    "The task executor: {} host is empty, cannot take-over, this might caused by the task hasn't dispatched",
+                    taskName);
+            return false;
+        }
 
+        final TaskExecutorReassignMasterRequest taskExecutorReassignMasterRequest =
+                TaskExecutorReassignMasterRequest.builder()
+                        .taskInstanceId(taskInstance.getId())
+                        .workflowHost(masterConfig.getMasterAddress())
+                        .build();
+        final TaskExecutorReassignMasterResponse taskExecutorReassignMasterResponse =
+                Clients
+                        .withService(IPhysicalTaskExecutorOperator.class)
+                        .withHost(taskInstance.getHost())
+                        .reassignWorkflowInstanceHost(taskExecutorReassignMasterRequest);
+        boolean success = taskExecutorReassignMasterResponse.isSuccess();
+        if (success) {
+            log.info("Reassign master host {} to {} successfully", taskExecutorHost, taskName);
+        } else {
+            log.info("Reassign master host {} on {} failed with response {}",
```

Comment:
```suggestion
            log.error("Reassign master host {} on {} failed with response {}",
```

Reply:
This is expected when worker failover, so the log level is info, we don't need to change to error, will cause alert if we have set alert for error log.